lib/redmine_goodies_macros.rb

```diff
@@ -0,0 +1,81 @@
+module RedmineGoodies
+    module Macros
+        # Retrieves a mapping of note references (that contain questions using the question macro) to their corresponding answers.
+        def self.get_all_answers(issue_id)
+            issue = Issue.find_by(id: issue_id)    
```

Comment:
I don't like that you need to make at least 2 queries to the database (one gets the issue, the other gets the journals ordered).

Isn't something disponible when the macro is run? E.g. `@journal`, `@...`?

The problem is that those queries will be done each time the macro is rendered.

Reply:
If you don't have that information, maybe get only the notes that have the answer macro in them. I'm wondering about the performance implication of a task with too many notes (e.g. > 1000)

Reply:
I don't want this macro to generate impact in the DB!

So the cross linking should be done via JS, client side.

Reply:
I moved the cross linking on client side (javascript)